Index: jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/com/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.groovyclient
+package com.jbehaveforjira.groovyclient
 import com.sun.jersey.api.client.Client
 import com.sun.jersey.api.client.ClientResponse
 import com.sun.jersey.api.client.WebResource
Index: jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/com/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.groovyclient
+package com.jbehaveforjira.groovyclient
 import com.sun.jersey.api.client.Client
 import org.apache.commons.lang.Validate
 import org.codehaus.jackson.map.ObjectMapper
Index: jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/com/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.groovyclient
+package com.jbehaveforjira.groovyclient
 
 import com.sun.jersey.api.client.Client
 import com.sun.jersey.api.client.WebResource
Index: jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/org/bitbucket/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/groovy/com/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy	(revision )
@@ -1,8 +1,8 @@
-package org.bitbucket.jbehaveforjira.groovyclient
+package com.jbehaveforjira.groovyclient
 import com.sun.jersey.api.client.Client
 import com.sun.jersey.api.client.ClientResponse
 import com.sun.jersey.api.client.WebResource
-import org.bitbucket.jbehaveforjira.javaclient.StoryPathsFinder
+import com.jbehaveforjira.javaclient.StoryPathsFinder
 
 import javax.ws.rs.core.MediaType
 import javax.ws.rs.core.Response
Index: jbehave-for-jira/jbehave-for-jira-groovy-client/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/pom.xml	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/pom.xml	(revision )
@@ -3,20 +3,19 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>org.bitbucket.pristrom</groupId>
-    <artifactId>jbehave-for-jira-groovy-client</artifactId>
+    <groupId>com.jbehaveforjira</groupId>
+    <artifactId>groovy-client</artifactId>
     <packaging>jar</packaging>
     <name>JBehave For Jira - Groovy Client</name>
     <version>1.0.0-SNAPSHOT</version>
     <description>Groovy client for integrating with the 'JBehave for Jira - Jira Plugin' component</description>
-    <url>https://bitbucket.org/pristrom/jbehave-for-jira-java-client-impl/overview</url>
+    <url>http://www.jbehaveforjira.com</url>
     <build>
-        <!--<resources>-->
-            <!--<resource>-->
-                <!--<directory>src/main/groovy</directory>-->
-            <!--</resource>-->
-        <!--</resources>-->
-        <sourceDirectory>src/main/groovy</sourceDirectory>
+        <resources>
+            <resource>
+                <directory>src/main/groovy</directory>
+            </resource>
+        </resources>
         <plugins>
             <plugin>
                 <groupId>org.codehaus.gmavenplus</groupId>
@@ -26,7 +25,6 @@
                     <execution>
                         <goals>
                             <goal>compile</goal>
-                            <goal>testCompile</goal>
                         </goals>
                         <configuration>
                             <sources>
@@ -37,6 +35,7 @@
                                     </includes>
                                 </source>
                             </sources>
+                            <outputDirectory>target/compiled-groovy</outputDirectory>
                         </configuration>
                     </execution>
                 </executions>
@@ -117,8 +116,8 @@
     </build>
     <dependencies>
         <dependency>
-            <groupId>org.bitbucket.pristrom</groupId>
-            <artifactId>jbehave-for-jira-java-client</artifactId>
+            <groupId>com.jbehaveforjira</groupId>
+            <artifactId>java-client</artifactId>
             <version>1.0.0-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
\ No newline at end of file
