Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Currency.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Currency.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/bookstore/Currency.java	(revision )
@@ -1,4 +1,4 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.bookstore;
 
 public enum Currency {
 
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Genre.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Genre.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/bookstore/Genre.java	(revision )
@@ -1,4 +1,4 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.bookstore;
 
 /**
  * @author Maryna Pristrom
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/FXType.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/FXType.java	(revision )
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/FXType.java	(revision )
@@ -0,0 +1,8 @@
+package com.jbehaveforjira.exampleproject.transaction;
+
+public enum FXType {
+    SPOT,
+    FORWARD,
+    SWAP
+
+}
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/BookRelatedSteps.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/BookRelatedSteps.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/BookRelatedSteps.java	(revision )
@@ -1,5 +1,7 @@
 package com.jbehaveforjira.exampleproject;
 
+import com.jbehaveforjira.exampleproject.bookstore.Book;
+import com.jbehaveforjira.exampleproject.bookstore.BookStore;
 import org.jbehave.core.annotations.Alias;
 import org.jbehave.core.annotations.Given;
 import org.jbehave.core.annotations.Named;
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Book.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Book.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/bookstore/Book.java	(revision )
@@ -1,4 +1,4 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.bookstore;
 
 import com.jbehaveforjira.javaclient.ParameterInfo;
 import org.jbehave.core.annotations.AsParameters;
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/BuySellIndicator.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/BuySellIndicator.java	(revision )
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/BuySellIndicator.java	(revision )
@@ -0,0 +1,6 @@
+package com.jbehaveforjira.exampleproject.transaction;
+
+public enum BuySellIndicator {
+    Buy,
+    Sell
+}
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/TransactionSteps.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/TransactionSteps.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/transaction/FxTransaction.java	(revision )
@@ -1,45 +1,14 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.transaction;
 
+import com.jbehaveforjira.exampleproject.bookstore.Currency;
 import com.jbehaveforjira.javaclient.ParameterInfo;
 import org.jbehave.core.annotations.AsParameters;
-import org.jbehave.core.annotations.Given;
 import org.jbehave.core.annotations.Parameter;
 
 import java.util.Date;
 
-public class TransactionSteps {
-
-    @Given("an FX transaction to buy $amount of $currency")
-    public void givenFxTransaction(@ParameterInfo(suggestedValues = {"100", "200", "300", "1000", "10000"}) Double amount, Currency currency) {
-        //
-    }
-
-    @Given("an FX transaction to sell $sellCurrency and buy $buyCurrency")
-    public void givenBuySellTransaction(Currency sellCurrency, Currency buyCurrency) {
-        //
-    }
-
-    @Given("an FX transaction with the following fields: $transactionFields")
-    public void givenFxTransaction(FxTransaction fxTransaction) {
-
-    }
-
-}
-
-enum FXType {
-    SPOT,
-    FORWARD,
-    SWAP
-
-}
-
-enum BuySellIndicator {
-    Buy,
-    Sell
-}
-
 @AsParameters
-class FxTransaction {
+public class FxTransaction {
 
     @Parameter(name = "Trader name")
     @ParameterInfo(suggestedValues = {"Andrew", "George", "Susan", "Helen"})
@@ -64,10 +33,10 @@
     Currency sellCurrency;
 
     @Parameter(name = "rate")
-    double rate;
+    Double rate;
 
     @Parameter(name = "Amount")
-    double amount;
+    Double amount;
 
     @Parameter(name = "Tx. Date")
     Date transactionDate;
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/BookStore.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/BookStore.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/bookstore/BookStore.java	(revision )
@@ -1,4 +1,4 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.bookstore;
 
 import java.util.ArrayList;
 import java.util.Collection;
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Format.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/Format.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/com/jbehaveforjira/exampleproject/bookstore/Format.java	(revision )
@@ -1,4 +1,4 @@
-package com.jbehaveforjira.exampleproject;
+package com.jbehaveforjira.exampleproject.bookstore;
 
 /**
  * @author Maryna Pristrom
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/TransactionSteps.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/TransactionSteps.java	(date 1411412403000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/com/jbehaveforjira/exampleproject/TransactionSteps.java	(revision )
@@ -1,12 +1,10 @@
 package com.jbehaveforjira.exampleproject;
 
+import com.jbehaveforjira.exampleproject.bookstore.Currency;
+import com.jbehaveforjira.exampleproject.transaction.FxTransaction;
 import com.jbehaveforjira.javaclient.ParameterInfo;
-import org.jbehave.core.annotations.AsParameters;
 import org.jbehave.core.annotations.Given;
-import org.jbehave.core.annotations.Parameter;
 
-import java.util.Date;
-
 public class TransactionSteps {
 
     @Given("an FX transaction to buy $amount of $currency")
@@ -26,58 +24,3 @@
 
 }
 
-enum FXType {
-    SPOT,
-    FORWARD,
-    SWAP
-
-}
-
-enum BuySellIndicator {
-    Buy,
-    Sell
-}
-
-@AsParameters
-class FxTransaction {
-
-    @Parameter(name = "Trader name")
-    @ParameterInfo(suggestedValues = {"Andrew", "George", "Susan", "Helen"})
-    String traderName;
-
-    @Parameter(name = "Counterparty")
-    String counterpartyName;
-
-    @Parameter(name = "Counterparty BIC")
-    String counterpartyBic;
-
-    @Parameter(name = "FX Type")
-    FXType fxType;
-
-    @Parameter(name = "Buy/Sell")
-    BuySellIndicator buyOrSellIndicator;
-
-    @Parameter(name = "Buy Currency")
-    Currency buyCurrency;
-
-    @Parameter(name = "Sell Currency")
-    Currency sellCurrency;
-
-    @Parameter(name = "rate")
-    double rate;
-
-    @Parameter(name = "Amount")
-    double amount;
-
-    @Parameter(name = "Tx. Date")
-    Date transactionDate;
-
-    @Parameter(name = "Sett. Date")
-    Date settlementDate;
-
-    @Parameter(name = "Account")
-    String account;
-
-    @Parameter(name = "Reference")
-    Long reference;
-}
