Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/ParameterInfo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/ParameterInfo.java	(date 1410985046000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/ParameterInfo.java	(revision )
@@ -25,6 +25,8 @@
     Class<? extends Enum> suggestedValuesEnum() default EmptyEnum.class;
 
     String formatPattern() default "";
+
+    String formatDisplayText() default "";
 }
 
 
