Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Format.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Format.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Format.java	(revision )
@@ -1,7 +1,7 @@
 package org.bitbucket.jbehaveforjira.exampleproject;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public enum Format {
 
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookStoreTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookStoreTest.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookStoreTest.java	(revision )
@@ -20,7 +20,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class BookStoreTest extends JUnitStories {
 
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/BookStore.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/BookStore.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/BookStore.java	(revision )
@@ -5,7 +5,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class BookStore {
 
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookRelatedSteps.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookRelatedSteps.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/test/java/org/bitbucket/jbehaveforjira/exampleproject/BookRelatedSteps.java	(revision )
@@ -8,7 +8,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class BookRelatedSteps {
 
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Book.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Book.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Book.java	(revision )
@@ -7,7 +7,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 @AsParameters
 public class Book {
Index: jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Genre.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Genre.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-example-project/src/main/java/org/bitbucket/jbehaveforjira/exampleproject/Genre.java	(revision )
@@ -1,7 +1,7 @@
 package org.bitbucket.jbehaveforjira.exampleproject;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public enum Genre {
     Fiction,
