Index: jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy	(date 1414444952000)
+++ jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy	(revision )
@@ -5,7 +5,6 @@
 import freemarker.ext.beans.BeansWrapper
 import freemarker.template.TemplateHashModel
 import freemarker.template.TemplateModelException
-import groovy.json.StringEscapeUtils
 import org.apache.commons.lang.StringEscapeUtils
 import org.apache.commons.lang.StringUtils
 import org.apache.commons.lang.Validate
Index: jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy	(date 1414444952000)
+++ jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStepDocReporter.groovy	(revision )
@@ -1,0 +1,0 @@
Index: jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy	(date 1414444952000)
+++ jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryReporter.groovy	(revision )
@@ -1,0 +1,0 @@
Index: jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy	(date 1414444952000)
+++ jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryLoader.groovy	(revision )
@@ -1,0 +1,0 @@
Index: jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy	(date 1414444952000)
+++ jbehave-for-jira/sync_copy/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyStoryPathsFinder.groovy	(revision )
@@ -1,0 +1,0 @@
