Index: jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy	(date 1414057528000)
+++ jbehave-for-jira/jbehave-for-jira-groovy-client/src/main/java/com/jbehaveforjira/groovyclient/GroovyTemplateStoryReporter.groovy	(revision )
@@ -54,7 +54,7 @@
 
     public GroovyTemplateStoryReporter(File file, Keywords keywords,
                                        String jiraBaseUrl, String jiraProject, String username, String password, String environment) {
-        this(file, keywords, new FreemarkerProcessor(), "ftl/jbehave-html-output.ftl", jiraBaseUrl, jiraProject, username, password, environment);
+        this(file, keywords, new FreemarkerProcessor(), "ftl/jira-jbehave-html-output.ftl", jiraBaseUrl, jiraProject, username, password, environment);
     }
 
     public GroovyTemplateStoryReporter(File file, Keywords keywords, TemplateProcessor processor, String templatePath,
