Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryLoader.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryLoader.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryLoader.java	(revision )
@@ -17,7 +17,7 @@
 import java.net.URISyntaxException;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class JiraStoryLoader implements StoryLoader {
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/ParameterInfo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/ParameterInfo.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/ParameterInfo.java	(revision )
@@ -5,7 +5,7 @@
 /**
  * Used to provide meta information for the JBehave parameter fields.
  *
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.FIELD, ElementType.PARAMETER})
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryPathsFinder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryPathsFinder.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryPathsFinder.java	(revision )
@@ -4,7 +4,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class JiraStoryPathsFinder implements StoryPathsFinder {
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/pom.xml	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/pom.xml	(revision )
@@ -24,7 +24,7 @@
     <developers>
         <developer>
             <id>pristrom</id>
-            <name>Maryna Stasyuk</name>
+            <name>Maryna Pristrom</name>
             <email>m.pristrom@gmail.com</email>
         </developer>
     </developers>
\ No newline at end of file
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/StoryPathsFinder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/StoryPathsFinder.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/StoryPathsFinder.java	(revision )
@@ -3,7 +3,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public interface StoryPathsFinder {
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStepDocReporter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStepDocReporter.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStepDocReporter.java	(revision )
@@ -6,7 +6,7 @@
 import java.util.List;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class JiraStepDocReporter implements StepdocReporter {
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/GroovyResourceHelper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/GroovyResourceHelper.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/GroovyResourceHelper.java	(revision )
@@ -12,7 +12,7 @@
 import java.net.URISyntaxException;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 class GroovyResourceHelper {
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryReporter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryReporter.java	(date 1410286683000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryReporter.java	(revision )
@@ -7,7 +7,7 @@
 import java.util.Map;
 
 /**
- * @author Maryna Stasyuk
+ * @author Maryna Pristrom
  */
 public class JiraStoryReporter implements StoryReporter {
 
