Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryLoader.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryLoader.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/JiraStoryLoader.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import com.sun.jersey.api.client.Client;
 import com.sun.jersey.api.client.ClientResponse;
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStepDocReporter.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStepDocReporter.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/JiraStepDocReporter.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import org.jbehave.core.reporters.StepdocReporter;
 import org.jbehave.core.steps.Stepdoc;
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryPathsFinder.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryPathsFinder.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/JiraStoryPathsFinder.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import java.util.Collections;
 import java.util.List;
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/StoryPathsFinder.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/StoryPathsFinder.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/StoryPathsFinder.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import java.util.List;
 
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/GroovyResourceHelper.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/GroovyResourceHelper.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/GroovyResourceHelper.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import com.sun.jersey.api.client.Client;
 import com.sun.jersey.api.client.ClientResponse;
Index: jbehave-for-jira/jbehave-for-jira-java-client/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/pom.xml	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/pom.xml	(revision )
@@ -3,13 +3,13 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>org.bitbucket.pristrom</groupId>
-    <artifactId>jbehave-for-jira-java-client</artifactId>
+    <groupId>com.jbehaveforjira</groupId>
+    <artifactId>java-client</artifactId>
     <packaging>jar</packaging>
     <name>JBehave For Jira - Java Client</name>
     <version>1.0.0-SNAPSHOT</version>
     <description>Java client for integrating with the 'JBehave for Jira - Jira Plugin' component</description>
-    <url>https://bitbucket.org/pristrom/jbehave-for-jira-java-client/wiki/Home</url>
+    <url>http://www.jbehaveforjira.com</url>
     <licenses>
         <license>
             <name>Apache License, Version 2.0</name>
@@ -18,8 +18,8 @@
         </license>
     </licenses>
     <scm>
-        <url>https://bitbucket.org/pristrom/jbehave-for-jira-java-client/src</url>
-        <connection>https://bitbucket.org/pristrom/jbehave-for-jira-java-client/src</connection>
+        <url>http://www.jbehaveforjira.com/java-client/src</url>
+        <connection>http://www.jbehaveforjira.com/java-client/src</connection>
     </scm>
     <developers>
         <developer>
\ No newline at end of file
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryReporter.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/JiraStoryReporter.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/JiraStoryReporter.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import org.jbehave.core.model.*;
 import org.jbehave.core.reporters.StoryReporter;
Index: jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/ParameterInfo.java
===================================================================
--- jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/org/bitbucket/jbehaveforjira/javaclient/ParameterInfo.java	(date 1410899865000)
+++ jbehave-for-jira/jbehave-for-jira-java-client/src/main/java/com/jbehaveforjira/javaclient/ParameterInfo.java	(revision )
@@ -1,4 +1,4 @@
-package org.bitbucket.jbehaveforjira.javaclient;
+package com.jbehaveforjira.javaclient;
 
 import java.lang.annotation.*;
 
@@ -24,7 +24,7 @@
 
     Class<? extends Enum> suggestedValuesEnum() default EmptyEnum.class;
 
-    String regExPattern() default "";
+    String formatPattern() default "";
 }
 
 
